saw-script/src/SAWScript/Import.hs

```diff
@@ -30,42 +33,49 @@ parseFile tokens = do
     Right stmts -> Right stmts
 
 -- | Load the 'Stmt's in a @.saw@ file.
-loadFile :: Options -> FilePath -> IO [Stmt]
+loadFile :: Options -> FilePath -> IO (Either [Text] [Stmt])
 loadFile opts fname = do
   printOutLn opts Info $ "Loading file " ++ show fname
   ftext <- TextIO.readFile fname
   -- XXX: the print functions should take care of printing the position
   -- (clean this up when we clean out the printing infrastructure)
-  -- XXX: the print functions should also take care of exiting on an error
-  -- (immediately or later). For now, throw errors and print anything else.
+  -- XXX: the print functions should also take care of exiting on an
+  -- error (immediately or later). For now, print warnings right away
+  -- and return errors in Either.
   case lexSAW fname ftext of
     Left (_, pos, txt) -> do
-      let txt' = show pos ++ ": " ++ Text.unpack txt
-      throwIO $ userError txt'
-
+      let txt' = Text.pack (show pos) <> ": " <> txt
+      return $ Left [txt']
     Right (tokens, optmsg) -> do
       case optmsg of
         Nothing -> return ()
+        -- The types make it possible to get an error back in the
+        -- optmsg field. Panic if we do, because the lexer isn't
+        -- supposed to do that, and currently does not.
+        Just (Error, _, _) ->
+          panic "loadFile" ["Lexer returned an error in the warning slot"]
         Just (vrb, pos, txt) -> do
           let txt' = show pos ++ ": " ++ Text.unpack txt
-          case vrb of
-            Error -> throwIO $ userError txt'
-            _ -> printOutLn opts vrb txt'
-      either throwIO return (parseFile tokens)
+          printOutLn opts vrb txt'
+      case parseFile tokens of
+        Left err -> return $ Left [Text.pack $ show err]
+        Right result -> return $ Right result
 
 -- | Find a file, potentially looking in a list of multiple search paths (as
 -- specified via the @SAW_IMPORT_PATH@ environment variable or
 -- @-i@/@--import-path@ command-line options). If the file was successfully
 -- found, load it. If not, raise an error.
-findAndLoadFile :: Options -> FilePath -> IO [Stmt]
+findAndLoadFile :: Options -> FilePath -> IO (Either [Text] [Stmt])
```

Comment:
A minor thing, but after these changes, this function no longer raises an error (as claimed in the Haddocks), but instead returns an error (via `Left`).